packages/quantic/force-app/main/default/lwc/quanticRefineModalContent/quanticRefineModalContent.js

```diff
@@ -99,21 +112,34 @@ export default class QuanticRefineModalContent extends LightningElement {
     this.headless = getHeadlessBundle(this.engineId);
     this.searchStatus = this.headless.buildSearchStatus(engine);
     this.breadcrumbManager = this.headless.buildBreadcrumbManager(engine);
+    this.getSortOptionsFromStore();
+
     this.unsubscribeSearchStatus = this.searchStatus.subscribe(() =>
-      this.gatherFacets()
+      this.getFacetDataFromStore()
     );
     this.unsubscribeBreadcrumbManager = this.breadcrumbManager.subscribe(() =>
       this.updateHasActiveFilters()
     );
   };
 
   /**
-   * Gather all facets registered in the Quantic store.
+   * Initializes all facets registered in the Quantic store.
    * @returns {void}
    */
-  gatherFacets() {
+  getFacetDataFromStore() {
     if (!this.hasFacets) {
-      this.data = getAllFacetsFromStore(this.engineId);
+      this.facetData = getAllFacetsFromStore(this.engineId);
+    }
+  }
+
+  /**
+   * Initializes all sort options registered in the Quantic store.
+   * @returns {void}
+   */
+  getSortOptionsFromStore() {
+    this.sortData = getAllSortOptionsFromStore(this.engineId);
+    if (this.sortData?.length) {
+      this.sortOptionsInitialized = true;
```

Comment:
The problem is here @SimonMilord, 

in the case where no  sort component was added in the search page, `this.sortOptionsInitialized = true;` will never happen which means the sort component will not be displayed.

While I think the logic that would make sense would be the following:

When the property `hideSort` is set to true:
- Don't display sort no matter what

When the property `hideSort` is set to false:
- if custom sort options found, display sort with custom sort options.
- if custom sort options not found, display sort with default options.

Reply:
as I mentioned yesterday, that's the behaviour we are aiming for:
```
- When the property hideSort is set to true:

     Don't display sort no matter what

- When the property hideSort is set to false:

    if custom sort options found, display sort with custom sort options.
    if custom sort options not found, display sort with default options.
```

Currently that's not entirely the case and that's an issue to fix in this component cause for example in the case where:
 
the `hideSort` is set to false and no sort options are found in the store, no sort component will be displayed while the expected behaviour is that a sort component with the default options will be displayed.




So this component needs to be updated, first in the HTML display the sort option based on a boolean `isSortComponentReady`:
```
    <template if:false={hideSort}>
      <div class="slds-var-m-bottom_small slds-clearfix">
        <template lwc:if={isSortComponentReady}>
          <c-quantic-sort class="slds-float_left" engine-id={engineId}>
            <template for:each={sortOptions} for:item="sortOption">
              <c-quantic-sort-option
                key={sortOption.value}
                value={sortOption.value}
                criterion={sortOption.criterion}
                label={sortOption.label}
                slot="sortOption"
              ></c-quantic-sort-option>
            </template>
          </c-quantic-sort>
        </template>
      </div>
    </template>
```

in the JS file, set `isSortComponentReady` after that you try to gather the sort options from the store:

```
  getSortOptionsFromStore() {
    this.sortData = getAllSortOptionsFromStore(this.engineId);
    this.isSortComponentReady = true;
  }
```